quilt3distribute/tests/test_dataset.py

```diff
@@ -171,3 +179,79 @@ def test_dataset_distribute(no_additions_dataset, extra_additions_dataset, push_
         mocked_package_push.return_value = "NiceTryGuy"
         no_additions_dataset.distribute(push_uri, "some message")
         extra_additions_dataset.distribute(push_uri, "some message")
+
+
+def test_dataset_metadata_numpy_type_casting(example_frame, example_readme):
+    # Add numpy column to frame
+    example_frame["NumpyTypes"] = np.zeros(9)
+    ds = Dataset(example_frame, "test_dataset", "me", example_readme)
+
+    # Add column filled with numpy types to index
+    ds.set_index_columns(["NumpyTypes"])
+
+    ds.distribute()
+
+
+class SomeDummyObject:
+    def __init__(self, a: int):
+        self.a = a
+
+
+def test_dataset_metadata_non_json_serializable_type(example_frame, example_readme):
+    # Add non json serializable type to dataframe
+    example_frame["BadType"] = [SomeDummyObject(i) for i in range(9)]
+    ds = Dataset(example_frame, "test_dataset", "me", example_readme)
+
+    # Add column filled with non serializable type to index
+    ds.set_index_columns(["BadType"])
+
+    # Check non json serializable type check fails
+    with pytest.raises(TypeError):
+        ds.distribute()
+
+
+def test_dataset_auto_metadata_grouping_no_additions(no_additions_dataset):
+    # Generate package
+    pkg = no_additions_dataset.distribute()
+
+    # Check file groupings available
+    assert set(pkg.keys()) == {
+        "2dReadPath", "3dReadPath", "MetadataReadPath", "README.md", "metadata.csv", "referenced_files"
+    }
+
+    # Check that associates was produced for metadata and nothing else
+    for f in pkg["2dReadPath"]:
+        assert "associates" in pkg["2dReadPath"][f].meta
+        assert len(pkg["2dReadPath"][f].meta) == 1
+
+
+def test_dataset_auto_metadata_grouping_extra_additions(extra_additions_dataset):
+    # Generate package
+    pkg = extra_additions_dataset.distribute()
+
+    # Check file groupings available
+    assert set(pkg.keys()) == {"MappedPath", "README.md", "metadata.csv", "referenced_files", "supporting_files"}
+
+    # Check that every metadata item is a string because no metadata reduction was needed
+    for f in pkg["MappedPath"]:
+        assert isinstance(pkg["MappedPath"][f].meta["Structure"], str)
+
+
+def test_dataset_auto_metadata_grouping_repeated_values(repeated_values_frame, example_readme):
```

Comment:
from the names and one line comments alone, it is a little difficult to figure out exactly what case you're testing. 
This is pedantic but maybe docstrings for the tests or more extensive statement of what exactly is testing.